spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfigurationTests.java

```diff
@@ -552,13 +553,61 @@ void subscriberExecutorProvider_globalAndSelectiveConfigurationSet_selectiveTake
         });
   }
 
+  @Test
+  void pullConfig_defaultConfigurationSet() {
```

Comment:
We should avoid setting defaults for the user, even if they are the same as the client library. There was at least one instance in which the client library changed the default, but Spring Cloud GCP did not, and it led to a hard-to-troubleshoot issue.
Best practice is to allow `null` in Properties, so that we only modify client library builders if the user explicitly set some option.

Reply:
I looked at original code, and it looks like we avoid setting our own defaults in `parallelPullCount` and `computeRetryableCodes`, but not for `computeMaxAckExtensionPeriod`, which you naturally used as a reference point. There was probably a reason for it, but see if you can manage to pass nulls back up to PublisherFactory for the new properties -- that would be better.

Reply:
Do you mean `SubscriberFactory` because new properties are subscriber settings. To double check, you think we should set the default value of two new properties to `null` rather than `0`?

Reply:
Sorry, yes, SubscriberFactory.
Leaving the value of wrapper types unset will automatically make them null. See if you can avoid providing a default value and allow the `null` to get returned through the layers.

Reply:
Done. 

Now the default value for `minDurationPerAckExtension` and `maxDurationPerAckExtension` is `null`.

I left the default value in docs as is (i.e., `0`). I'm not sure whether I should also change it to `null`, since most default values are not what they actually are.